components/Nav/Nav.test.tsx

```diff
@@ -0,0 +1,64 @@
+import { render, screen } from '@testing-library/react';
+import Nav from './Nav';
+
+const mockPush = jest.fn();
+const mockUsePathname = jest.fn();
+
+const mockUseAuthContext = {
+  logoutAccount: jest.fn(),
+};
+
+jest.mock('next/navigation', () => ({
+  useRouter() {
+    return {
+      push: mockPush,
+    };
+  },
+  usePathname() {
+    return mockUsePathname();
+  },
+}));
+
+jest.mock('../../context/AuthContextProvider', () => ({
+  useAuthContext() {
+    return {
+      ...mockUseAuthContext,
+    };
+  },
+}));
+
+describe('Nav', () => {
+  beforeAll(() => {
+    jest.clearAllMocks();
+  });
+  it('it should be hidden when path is /register', () => {
```

Comment:
Q: I wonder if we can combine these visibility tests? They're very similar in what they're testing, but only the mock route is changing. Just a thought though.

Reply:
Yes, I know what you mean. It does seem redundant but the mock routes are all different from each other so not sure how to combine. Open to ideas though!